### PR TITLE
Add wall-drawing GUI

### PR DESCRIPTION
**Pull request by beini19**
- Added basic version of wall-selector for the A* module
- Currently slow when running alongside the image viewer, needs optimization

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -36,7 +36,32 @@ find_package(Threads REQUIRED)
 
 if (Qt5_POSITION_INDEPENDENT_CODE)
     set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-endif ()
+endif()
+
+## Third Party files
+set(THIRD_PARTY_COMMON
```
Comment (Mogball) [inline CMakeLists.txt]: All your PRs have this weird bunch of old code; I think you fork's master branch may be out-of-date

File: code/gui/griddisplay.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    void buttonClicked(QPushButton *, int x, int y);
+    void hideGrid();
+
+private:
+    void showView();
+    void updateScene();
+    void drawGrid();
+    void drawButtons();
+
+    QGraphicsScene *m_scene;
```
Comment (Mogball) [inline code/gui/griddisplay.h]: if you can change these to `std::unique_ptr`s are well it'd be fantastique

File: code/gui/griddisplay.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    QGraphicsView *m_view;
+    QPushButton *m_button[40][20];  //TODO: Replace hardcoded values
+    bool squareSelected[40][20];
+    //QSignalMapper *m_signalmapper;
+
+
+    QString buttonStyle = QString (
```
Comment (Mogball) [inline code/gui/griddisplay.h]: I don't think you need the `QString` in front of this

File: code/gui/griddisplay.h [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+private:
+    void showView();
+    void updateScene();
+    void drawGrid();
+    void drawButtons();
+
+    QGraphicsScene *m_scene;
+    QGraphicsView *m_view;
+    QPushButton *m_button[40][20];  //TODO: Replace hardcoded values
+    bool squareSelected[40][20];
```
Comment (Mogball) [inline code/gui/griddisplay.h]: use `utility/array2d.h`

**Comment by Mogball** [comment]
also I'm not actually running into any performance issues on my end o.o

**Comment by beini19** [comment]
My camera is 720p, although the buttons should only be drawn once so I'm not sure if that's the issue. Let's test it with the controls computer/camera at the next lab meeting.

**Pull request merged**